extra/import_data.py

```diff
@@ -1,197 +1,158 @@
+    if lastUpdatedOn_mongoDB is None:
+        db.replace_one({"version" : 1}, {"lastUpdatedOn" : '2018-01-01T00:00:00.000+0000', "version" : 1}, True)
+        lastUpdatedOn_mongoDB = db.find_one({"version" : 1}, {"lastUpdatedOn":1, "_id":0})
+    
+    for key, value in lastUpdatedOn_mongoDB.items():
+        lastUpdatedOn_mongoDB = value
+    
+    # retrieving data from Alissa
+    analysis = None
+    for analysis in client.get_analyses(status='COMPLETED', 
+                                        lastUpdatedAfter=lastUpdatedOn_mongoDB, # format is '2020-01-01T00:00:00.000+0000'
+                                        analysisPipelineName='ONB01',
+                                        analysisType='INHERITANCE'): #add for testing [:x], where x is numer of iterations
+        if not analysis['classificationTreeName']:  # Skip analysis
+            continue
+        else:
+            # retrieve basic info from Alissa about the analysis
+            patient_dn_no = client.get_report(analysis['id'])[0]['reportName'].split("_")[0]
+            logger.info('start Alissa retrieval of: %s' % patient_dn_no)
+            ONB01_analysis = client.get_inheritance_analyses(analysis['id'])
+            accession_number = client.get_patient(analysis['patientId'])['accessionNumber']
+            export_id = client.post_inheritance_analyses_variants_export(analysis['id'], marked_review=True, marked_include_report=False)['exportId']
+            analyis_sources = client.get_sources(analysis['id'])
+            
+            # retrieve the VUS/GUS info based on the analysis ID
+            VUS_response = None
+            while VUS_response is None:
+                try:
+                    time.sleep(5)  # 5 sec delay to request exported report.
+                    VUS_response = client.get_inheritance_analyses_variants_export(analysis['id'], export_id)
+                    #response.raise_for_status()
+                except HTTPError as exception:
+                    print(exception)
+                    pass
+            logger.info('Alissa retrieval completed of: %s' % patient_dn_no)
+            # sometimes the there are no VUS marked or found within an analysis, then no info needs to be uploaded
+            if VUS_response == []:
+                logger.info('Patient %s, has no VUS marked/found. Not uploaded to MongoDB' % patient_dn_no)
+                continue
+            else:
+                upload_to_mongodb(ONB01_analysis, accession_number, analyis_sources, patient_dn_no, VUS_response)
+    # if latest date does not retrieve any analyses from Alissa, quit program and try later
+    if analysis is None:
+        logger.info('no new analysis in Alissa since last upload to VUSualizer')
+        sys.exit(0)
+    # after uploading all analyses to MongoDB, replace time of last upload with starttime of script (before calling Alissa)
+    db.replace_one({"version" : 1}, {"lastUpdatedOn" : start_time, "version" : 1}, True)
```

Comment:
Why do you not want to update the `lastUpdatedOn` when there are no analysis found?

Reply:
keeps the question asked to the Alissa database smaller, but maybe a bit overkill?
It is also to see in the mongoDB itself, when it was lasted checked.

Reply:
I read your comment the wrong way I see now.
I thought about it. I thought it might be a bit on the safer side to just check again the next day, but it also seems necessary. will change it 👍